src/Features/Core/Portable/UseCoalesceExpression/AbstractUseCoalesceExpressionDiagnosticAnalyzer.cs

```diff
@@ -95,12 +97,30 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
             }
 
             if (!syntaxFacts.AreEquivalent(
-                    conditionRightIsNull ? conditionLeftLow : conditionRightLow, 
+                    conditionRightIsNull ? conditionLeftLow : conditionRightLow,
                     isEquals ? whenFalseNodeLow : whenTrueNodeLow))
             {
                 return;
             }
 
+            var semanticModel = context.SemanticModel;
+            var conditionType = semanticModel.GetTypeInfo(
+                conditionLeftIsNull ? conditionRightLow : conditionLeftLow, cancellationTokan).Type;
+            if (conditionType != null &&
```

Comment:
In general our approach is to offer during error conditions so that people can use the features even when they're in the middle of doing changes.  The expectation then is that it's *more* ok to do somtehing bad because their code *already* had squiggles in it to begin with letting you know that there are issues.